### ISSUE TITLE
conda 22.9.0 breaks bash command prompt

### ISSUE THREAD

**Issue by SimonPeatman** [conda/conda#11885]
### What happened?

I have upgraded from 4.14.0 to 22.9.0 and now my bash command prompt is broken.  I used to see:
```(conda-env) hostname:/path/to/working/dir$ ```

But now I just get:
```(conda-env) ```

In other words, rather than prepending the name of the conda environment in parentheses to the value of the PS1 environment variable, conda is overwriting PS1 altogether.  Doing ```conda deactivate``` leaves PS1 as an empty string.

### The Fix

Run the following and restart your shell for the changes to take effect:

```
conda init bash
```

See: https://github.com/conda/conda/issues/11885#issuecomment-1259519190

### Conda Details

<details>
<summary><code>conda info</code></summary>

```shell
active environment : himawari
    active env location : /home/users/peatman/miniconda3/envs/himawari
            shell level : 2
       user config file : /home/users/peatman/.condarc
 populated config files : /home/users/peatman/.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.9.7.final.0
       virtual packages : __linux=3.10.0=0
                          __glibc=2.17=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/users/peatman/miniconda3  (writable)
      conda av data dir : /home/users/peatman/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/users/peatman/miniconda3/pkgs
                          /home/users/peatman/.conda/pkgs
       envs directories : /home/users/peatman/miniconda3/envs
                          /home/users/peatman/.conda/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.9.7 Linux/3.10.0-1160.66.1.el7.x86_64 centos/7.9.2009 glibc/2.17
                UID:GID : 7050034:26030
             netrc file : /home/users/peatman/.netrc
           offline mode : False
```

</details>
<details>
<summary><code>conda config</code></summary>

```shell
==> /home/users/peatman/.condarc <==
auto_activate_base: False
```

</details>
<details>
<summary><code>conda list</code></summary>

```shell
# packages in environment at /home/users/peatman/miniconda3/envs/himawari:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
boost-cpp                 1.70.0               ha2d47e9_1    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.18.1               h7f98852_0    conda-forge
ca-certificates           2022.07.19           h06a4308_0    defaults
cairo                     1.16.0               h19f5f5c_2    defaults
certifi                   2022.9.14        py39h06a4308_0    defaults
cfitsio                   3.470                hb418390_7    conda-forge
curl                      7.84.0               h5eee18b_0    defaults
expat                     2.2.10               h9c3ff4c_0    conda-forge
fontconfig                2.13.1               h6c09931_0    defaults
freetype                  2.10.4               h0708190_1    conda-forge
freexl                    1.0.6                h7f98852_0    conda-forge
gdal                      3.4.1            py39h2c27f0e_0    defaults
geos                      3.8.0                he6710b0_0    defaults
geotiff                   1.7.0                hd69d5b1_0    defaults
giflib                    5.2.1                h36c2ea0_2    conda-forge
glib                      2.69.1               h4ff587b_1    defaults
hdf4                      4.2.13               h3ca952b_2    defaults
hdf5                      1.10.6          nompi_h6a2412b_1114    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
jpeg                      9e                   h166bdaf_1    conda-forge
json-c                    0.13.1            hbfbb72e_1002    conda-forge
kealib                    1.4.14               hb50703a_1    defaults
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.19.3               h3790be6_0    conda-forge
ld_impl_linux-64          2.38                 h1181459_1    defaults
libblas                   3.9.0           15_linux64_openblas    conda-forge
libcblas                  3.9.0           15_linux64_openblas    conda-forge
libcurl                   7.84.0               h91b91d3_0    defaults
libdap4                   3.19.1               h6ec2957_0    defaults
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  he6710b0_2    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgdal                   3.4.1                h05199a0_1    defaults
libgfortran-ng            12.1.0              h69a702a_16    conda-forge
libgfortran5              12.1.0              hdcd56e2_16    conda-forge
libgomp                   11.2.0               h1234567_1    defaults
libkml                    1.3.0             h4fcabce_1010    conda-forge
liblapack                 3.9.0           15_linux64_openblas    conda-forge
libmagic                  5.39                 h27826a3_0    conda-forge
libnetcdf                 4.8.1                h42ceab0_1    defaults
libnghttp2                1.46.0               hce63b2e_0    defaults
libopenblas               0.3.20          pthreads_h78a6416_0    conda-forge
libpng                    1.6.37               hbc83047_0    defaults
libpq                     12.9                 h16c4e8d_3    defaults
libspatialite             4.3.0a              hbedb2dc_20    defaults
libssh2                   1.10.0               ha56f1ee_2    conda-forge
libstdcxx-ng              11.2.0               h1234567_1    defaults
libtiff                   4.2.0                hecacb30_2    defaults
libuuid                   1.0.3                h7f8727e_2    defaults
libwebp-base              1.2.2                h7f8727e_0    defaults
libxcb                    1.15                 h7f8727e_0    defaults
libxml2                   2.9.14               h74e7548_0    defaults
libzip                    1.8.0                h5cef20c_0    defaults
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
ncurses                   6.3                  h5eee18b_3    defaults
numpy                     1.22.3           py39hc58783e_2    conda-forge
openjpeg                  2.3.1                hf7af979_3    conda-forge
openssl                   1.1.1q               h7f8727e_0    defaults
pcre                      8.45                 h9c3ff4c_0    conda-forge
pip                       22.1.2           py39h06a4308_0    defaults
pixman                    0.40.0               h36c2ea0_0    conda-forge
poppler                   0.81.0               h01f5e8b_2    defaults
poppler-data              0.4.11               hd8ed1ab_0    conda-forge
proj                      6.2.1                hc80f0dc_0    conda-forge
python                    3.9.13               haa1d7c7_1    defaults
python-magic              0.4.27           py39hf3d152e_2    conda-forge
python_abi                3.9                      2_cp39    conda-forge
readline                  8.1.2                h7f8727e_1    defaults
setuptools                63.4.1           py39h06a4308_0    defaults
sqlite                    3.39.2               h5082296_0    defaults
tiledb                    2.3.3                h1132f93_2    defaults
tk                        8.6.12               h1ccaba5_0    defaults
tzdata                    2022c                h04d1e81_0    defaults
wheel                     0.37.1             pyhd3eb1b0_0    defaults
xerces-c                  3.2.3                h780794e_0    defaults
xz                        5.2.5                h7f8727e_1    defaults
zlib                      1.2.12               h5eee18b_3    defaults
zstd                      1.5.2                ha4553b6_0    defaults
```

</details>

**Comment by travishathaway** [comment]
Hi @SimonPeatman,

Thanks for submitting this issue. Your are the very first for our latest release. I was trying to recreate this issue locally, but I was not able to. Would you mind sharing what your bash prompt looks like (or any other relevant details in your `.bashrc` or `.bash_profile`). This will help us further debug the issue.

**Comment by SimonPeatman** [comment]
Here's a prompt before conda fiddles with it:
```
login1.jasmin.ac.uk:/home/users/peatman$ 
```
And then:
```
login1.jasmin.ac.uk:/home/users/peatman$ conda activate himawari
(himawari) 
```
Whereas under previous versions of conda it would look like:
```
login1.jasmin.ac.uk:/home/users/peatman$ conda activate himawari
(himawari) login1.jasmin.ac.uk:/home/users/peatman$ 
```

My ```~/.bash_profile``` is just:
```
# Get the aliases and functions
if [ -f ~/.bashrc ]; then
        . ~/.bashrc
fi

# User specific environment and startup programs
PATH=$PATH:$HOME/bin
export PATH
```

Here's my ```~/.bashrc``` file:
```
# Source global definitions
if [ -f /etc/bashrc ]; then
	. /etc/bashrc
fi

if [[ $HOSTNAME = host*.jc.rl.ac.uk ]] || [[ $HOSTNAME = sci?.jasmin.ac.uk ]] || [[ $HOSTNAME = mass-cli?.ceda.ac.uk ]] ; then
    export TMDIR=/gws/nopw/j04/terramaris
    export PATH=$PATH:/home/users/peatman/bin:/home/users/peatman/bash:/home/users/peatman/convsh
    export PYTHONPATH=$PYTHONPATH:/home/users/peatman/python
    export LD_LIBRARY_PATH=$HOME/.local/lib:$LD_LIBRARY_PATH
fi

# User specific aliases and functions
if [ -f ~/.bash_aliases ]; then
	. ~/.bash_aliases
fi
if [ -f ~/.bash_functions ]; then
	. ~/.bash_functions
fi

# command line prompt
PS1='$HOSTNAME:$PWD\$ '

[[ $- != *i* ]] && return # Stop here if not running interactively
[[ $(hostname) = "jasmin-cylc.ceda.ac.uk" ]] && . mosrs-setup-gpg-agent
# Enable bash completion for Rose commands
[[ -f /apps/contrib/metomi/rose/etc/rose-bash-completion ]] && . /apps/contrib/metomi/rose/etc/rose-bash-completion

# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/home/users/peatman/miniconda3/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/home/users/peatman/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/home/users/peatman/miniconda3/etc/profile.d/conda.sh"
    else
        export PATH="/home/users/peatman/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<

[[ $HOSTNAME = host*.jc.rl.ac.uk ]] || [[ $HOSTNAME = sci?.jasmin.ac.uk ]] && conda activate py3.10
```

The file ```/etc/bashrc```, which is sourced at the start, does exist and it sets PS1, but my ```~/.bashrc``` overwrites PS1 after that anyway.

Incidentally, someone else has [raised exactly the same bug on Stack Overflow](https://stackoverflow.com/questions/73860482/changing-how-conda-activate-deactivate-updates-ps1).

**Comment by kenodegard** [comment]
@SimonPeatman what do you get when running:

```
PS1="${PS1:-}" conda shell.posix activate
```

**Comment by SimonPeatman** [comment]
@kenodegard 

> @SimonPeatman what do you get when running:
> 
> ```
> PS1="${PS1:-}" conda shell.posix activate
> ```

```
(himawari) PS1="${PS1:-}" conda shell.posix activate
. "/home/users/peatman/miniconda3/envs/himawari/etc/conda/deactivate.d/proj4-deactivate.sh"
. "/home/users/peatman/miniconda3/envs/himawari/etc/conda/deactivate.d/geotiff-deactivate.sh"
. "/home/users/peatman/miniconda3/envs/himawari/etc/conda/deactivate.d/gdal-deactivate.sh"
PS1='(base) '
export PATH='/home/users/peatman/miniconda3/bin:/home/users/peatman/miniconda3/condabin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin:/apps/contrib/metomi/bin:/home/users/peatman/bin:/home/users/peatman/bash:/home/users/peatman/convsh:/home/users/peatman/bin'
export CONDA_PREFIX='/home/users/peatman/miniconda3'
export CONDA_SHLVL='2'
export CONDA_DEFAULT_ENV='base'
export CONDA_PROMPT_MODIFIER='(base) '
export CONDA_PREFIX_1='/home/users/peatman/miniconda3/envs/himawari'
export CONDA_EXE='/home/users/peatman/miniconda3/bin/conda'
export _CE_M=''
export _CE_CONDA=''
export CONDA_PYTHON_EXE='/home/users/peatman/miniconda3/bin/python'
```

**Comment by kenodegard** [comment]
@SimonPeatman can you run that again before activating `conda` the first time?

**Comment by SimonPeatman** [comment]
@kenodegard 

> @SimonPeatman can you run that again before activating `conda` the first time?

```
sci1.jasmin.ac.uk:/home/users/peatman$ PS1="${PS1:-}" conda shell.posix activate
PS1='(base) '
export PATH='/home/users/peatman/miniconda3/bin:/home/users/peatman/miniconda3/condabin:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/puppetlabs/bin:/apps/contrib/metomi/bin:/home/users/peatman/bin:/home/users/peatman/bash:/home/users/peatman/convsh:/home/users/peatman/bin'
export CONDA_PREFIX='/home/users/peatman/miniconda3'
export CONDA_SHLVL='1'
export CONDA_DEFAULT_ENV='base'
export CONDA_PROMPT_MODIFIER='(base) '
export CONDA_EXE='/home/users/peatman/miniconda3/bin/conda'
export _CE_M=''
export _CE_CONDA=''
export CONDA_PYTHON_EXE='/home/users/peatman/miniconda3/bin/python'
sci1.jasmin.ac.uk:/home/users/peatman$ 
```

> And can you also try running ```conda init bash``` and then restarting your shell and see if that fixes it?


Yes, this fixes it! :)  I now have the correct prompt:
```
sci1.jasmin.ac.uk:/home/users/peatman$ conda activate himawari
(himawari) sci1.jasmin.ac.uk:/home/users/peatman$ 
```

**Comment by bingyao** [comment]
I also run into the same issue. After update to 22.9.0, my BASH shell prompt disappeared, only `(base)` left, and `PS1` is emptied.

So I just roll back to the previous revision and get the `PS1` back.

**Comment by nshoman** [comment]
I also had the same problem, but the `conda init bash` fix worked for me

**Comment by kenodegard** [comment]
When running `conda init bash`, what files are being modified?

**Comment by stockeh** [comment]
@kenodegard  
> What files are being modified when running `conda init bash`?

Take note of the `modified` files.

```
(dev) conda init bash
modified      ~/miniconda/condabin/conda
modified      ~/miniconda/bin/conda
modified      ~/miniconda/bin/conda-env
no change     ~/miniconda/bin/activate
no change     ~/miniconda/bin/deactivate
no change     ~/miniconda/etc/profile.d/conda.sh
no change     ~/miniconda/etc/fish/conf.d/conda.fish
no change     ~/miniconda/shell/condabin/Conda.psm1
no change     ~/miniconda/shell/condabin/conda-hook.ps1
no change     ~/miniconda/lib/python3.9/site-packages/xontrib/conda.xsh
no change     ~/miniconda/etc/profile.d/conda.csh
no change     ~/.bashrc
```

**Comment by kathoef** [comment]
I do have a potentially related problem, i.e. `conda init bash` executed by an unprivileged user does not succeed anymore... which causes one of my Docker container images to be broken / not to be build anymore. It seems like `conda init bash` behaviour has changed with the update from v4.12.0 to v22.9.0. Below is some output from an interactive shell session in an Ubuntu 22.04 latest container image.

```shell
root@d64e9d7a32e6:/# apt update && apt install wget --yes
root@d64e9d7a32e6:/# wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
root@d64e9d7a32e6:/# bash Miniconda3-latest-Linux-x86_64.sh -f -b -p /miniconda3
root@d64e9d7a32e6:/# source /miniconda3/etc/profile.d/conda.sh
root@d64e9d7a32e6:/# conda --version
conda 4.12.0
root@d64e9d7a32e6:/# conda init bash --dry-run
no change     /miniconda3/condabin/conda
no change     /miniconda3/bin/conda
no change     /miniconda3/bin/conda-env
no change     /miniconda3/bin/activate
no change     /miniconda3/bin/deactivate
no change     /miniconda3/etc/profile.d/conda.sh
no change     /miniconda3/etc/fish/conf.d/conda.fish
no change     /miniconda3/shell/condabin/Conda.psm1
no change     /miniconda3/shell/condabin/conda-hook.ps1
no change     /miniconda3/lib/python3.9/site-packages/xontrib/conda.xsh
no change     /miniconda3/etc/profile.d/conda.csh
modified      /root/.bashrc
==> For changes to take effect, close and re-open your current shell. <==
root@d64e9d7a32e6:/# conda update conda --yes
root@d64e9d7a32e6:/# conda --version
conda 22.9.0
root@d64e9d7a32e6:/# conda init bash --dry-run
modified      /miniconda3/condabin/conda
modified      /miniconda3/bin/conda
modified      /miniconda3/bin/conda-env
no change     /miniconda3/bin/activate
no change     /miniconda3/bin/deactivate
no change     /miniconda3/etc/profile.d/conda.sh
no change     /miniconda3/etc/fish/conf.d/conda.fish
no change     /miniconda3/shell/condabin/Conda.psm1
no change     /miniconda3/shell/condabin/conda-hook.ps1
no change     /miniconda3/lib/python3.9/site-packages/xontrib/conda.xsh
no change     /miniconda3/etc/profile.d/conda.csh
modified      /root/.bashrc
==> For changes to take effect, close and re-open your current shell. <==
```
The changes attempted by `conda init bash` are the same for an unprivileged user, which of course gives an error. (The changes to `/home/jovyan/.bashrc` are correctly applied, though.)

```
root@d64e9d7a32e6:/# useradd --uid 1000 --shell /bin/bash --create-home jovyan
root@d64e9d7a32e6:/# su jovyan
jovyan@d64e9d7a32e6:/$ conda init bash --dry-run
modified      /miniconda3/condabin/conda
modified      /miniconda3/bin/conda
modified      /miniconda3/bin/conda-env
no change     /miniconda3/bin/activate
no change     /miniconda3/bin/deactivate
no change     /miniconda3/etc/profile.d/conda.sh
no change     /miniconda3/etc/fish/conf.d/conda.fish
no change     /miniconda3/shell/condabin/Conda.psm1
no change     /miniconda3/shell/condabin/conda-hook.ps1
no change     /miniconda3/lib/python3.9/site-packages/xontrib/conda.xsh
no change     /miniconda3/etc/profile.d/conda.csh
modified      /home/jovyan/.bashrc
==> For changes to take effect, close and re-open your current shell. <==
jovyan@d64e9d7a32e6:/$ conda init bash                 

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/miniconda3/lib/python3.9/site-packages/conda/exceptions.py", line 1129, in __call__
        return func(*args, **kwargs)
      File "/miniconda3/lib/python3.9/site-packages/conda/cli/main.py", line 86, in main_subshell
        exit_code = do_call(args, p)
      File "/miniconda3/lib/python3.9/site-packages/conda/cli/conda_argparse.py", line 93, in do_call
        return getattr(module, func_name)(args, parser)
      File "/miniconda3/lib/python3.9/site-packages/conda/cli/main_init.py", line 50, in execute
        return initialize(context.conda_prefix, selected_shells, for_user, args.system,
      File "/miniconda3/lib/python3.9/site-packages/conda/core/initialize.py", line 119, in initialize
        run_plan_elevated(plan2)
      File "/miniconda3/lib/python3.9/site-packages/conda/core/initialize.py", line 688, in run_plan_elevated
        result = subprocess_call(
      File "/miniconda3/lib/python3.9/site-packages/conda/gateways/subprocess.py", line 89, in subprocess_call
        process = Popen(
      File "/miniconda3/lib/python3.9/subprocess.py", line 951, in __init__
        self._execute_child(args, executable, preexec_fn, close_fds,
      File "/miniconda3/lib/python3.9/subprocess.py", line 1821, in _execute_child
        raise child_exception_type(errno_num, err_msg, err_filename)
    FileNotFoundError: [Errno 2] No such file or directory: 'sudo'

`$ /miniconda3/bin/conda init bash`

  environment variables:
                 CIO_TEST=<not set>
                CONDA_EXE=/miniconda3/bin/conda
         CONDA_PYTHON_EXE=/miniconda3/bin/python
               CONDA_ROOT=/miniconda3
              CONDA_SHLVL=0
           CURL_CA_BUNDLE=<not set>
                     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/game
                          s:/usr/local/games:/snap/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : None
            shell level : 0
       user config file : /home/jovyan/.condarc
 populated config files : 
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.9.12.final.0
       virtual packages : __linux=5.10.124=0
                          __glibc=2.35=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /miniconda3  (read only)
      conda av data dir : /miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /miniconda3/pkgs
                          /home/jovyan/.conda/pkgs
       envs directories : /home/jovyan/.conda/envs
                          /miniconda3/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.9.12 Linux/5.10.124-linuxkit ubuntu/22.04.1 glibc/2.35
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

If submitted, this report will be used by core maintainers to improve
future releases of conda.
Would you like conda to send this report to the core maintainers? [y/N]: N

No report sent. To permanently opt-out, use

    $ conda config --set report_errors false
```

The latter worked perfectly with conda v4.12.0.

**Comment by ekungurov** [comment]
What 'conda init bash' does exactly? 
Could it be fixed so that re-initialization won't be needed?

**Comment by matifali** [comment]
this breaks conda in Dockerfiles

**Comment by xaverm** [comment]
Just had the same issue (with a per-user installation) after `conda update -n base -c conda-forge conda` from `4.14` to `22.9`. The  `conda init bash` fix worked for me.

**Comment by matifali** [comment]
> Just had the same issue (with a per-user installation) after `conda update -n base -c conda-forge conda` from `4.14` to `22.9`. The  `conda init bash` fix worked for me.

Is it working? Is it working for you in a Docker file?

**Comment by davidecarlson** [comment]
Same thing happened to me on a CentOS 7.x HPC cluster.  Running `conda init bash` does indeed fix the issue, but I am not going to make all the users of the cluster do that.

Currently in the process of trying to revert back to to conda 4.14

**Comment by jaimergp** [comment]
The issue comes from `$CONDA/bin/conda` having a different shebang in this release (unfortunate side-effect of a different fix, you can learn the details here: https://github.com/conda/conda/issues/11893).

`conda init bash` works because it rewrites the entry point with a normal shebang. If this command cannot be executed for different reasons (permissions, inconvenient, etc), one can apply a temporary fix by changing the shebang manually.

From:

```
#!/bin/sh
'''exec' "/path/where/conda/is/installed/bin/python" "$0" "$@"
' '''
# -*- coding: utf-8 -*-
...
```

To:

```
#!/path/where/conda/is/installed/bin/python
# -*- coding: utf-8 -*-
...
```

Affected files are:

* ./condabin/conda
* ./bin/conda
* ./bin/conda-env


Sorry for the disruption! A fix is being worked on!

**Comment by tlunet** [comment]
Hi all 

(hope I'm writing in the main issue about this problem :sweat_smile:)

I got the same issue on an OpenSuse system (bash command prompt broken), but **not** on my Ubuntu 20.04 laptop that I use in parallel ... which was weird. After searching for a while, I may have found where the issue comes from ...

There has been a particular change in 22.9.0 in `conda/core/portability.py` : 

https://github.com/conda/conda/blob/60e11b4f1859874c1cb1a3ae324e599e0867a086/conda/core/portability.py#L208-L230

which basically transforms the `conda` executable header from

```py
#!/prog/anaconda/bin/python
# -*- coding: utf-8 -*-
import sys
...
```

to 

```py
#!/bin/sh
'''exec' "/prog/anaconda/bin/python" "$0" "$@"
' '''
# -*- coding: utf-8 -*-
import sys
...
```

_here `/prog/anaconda/` is my personal installation path_

If I'm right, the idea came from #11676 and made possible to run the conda executable with both `python` or `sh`. This is run using `sh` during conda activation, by providing the `PS1` environment variable (used for the prompt) to the script, by evaluating a shell command like this :

```bash
PS1=$PS1 /prog/anaconda/bin/conda shell.posix activate
```

It is supposed to pass the value of the `PS1` environment variable to the python program of conda, so it can get it back using `os.environ` and prepend the `(base)` prefix to it. Problem is, `sh` may not pass the value of the `PS1` variable depending on the system. For instance, on my Ubuntu 20.04 system (where prompt is not broken), it passes the `PS1` value, but on my OpenSuse computer, it does not (and prompt is broken).

You can quickly test it on you system with the following script 

```python
#!/bin/sh
'''exec' "/prog/anaconda/bin/python" "$0" "$@"
' '''
# -*- coding: utf-8 -*-
import os
print(os.environ['PS1'])
```

--> make it executable with `chmod +x` and run it like this :

```bash
PS1="blabla" ./test
```

if you get and error, it means that `PS1` value is not passed, while if this works, then it should print "blabla" in your console.

Apparently, there is some differences between the sh executable in my system. While on my OpenSuse (broken prompt), I get :

```bash
$ /usr/bin/sh --version
GNU bash, Version 4.4.23(1)-release (x86_64-suse-linux-gnu)
Copyright (C) 2016 Free Software Foundation, Inc.
Lizenz GPLv3+: GNU GPL Version 3 oder jünger <http://gnu.org/licenses/gpl.html>

Dies ist freie Software.  Sie darf verändert und verteilt werden.
Es wird keine Garantie gewährt, soweit das Gesetz es zulässt.
```

on my Ubuntu system ... well the `--version` argument does not work with `sh` :sweat_smile: 

One solution I found to make it work on OpenSuse, was to add the "interactive" option to `sh`, adding the `-i` option in the script header like this :

```python
#!/bin/sh -i
'''exec' "/prog/anaconda/bin/python" "$0" "$@"
' '''
# -*- coding: utf-8 -*-
import os
print(os.environ['PS1'])
```

This works on both systems, and doing the same modification on the `conda` executable (located in `/prog/anaconda/bin/conda`) fixed the prompt. However, doing the same modification in `conda` executable for my Ubuntu 20.04 system did added a lot of console output of the `PS1` value when activating the conda base environment during the start of a new terminal session.

At then end I'm not sure how this could be solved, so conda 22.9.0 would work on both systems. But my noob guess would be to look into that direction ... :upside_down_face:

**Comment by tlunet** [comment]
Oups, sorry ... just saw the previous comment

**Comment by jaimergp** [comment]
@tlunet heh you reached to the same conclusion we did in https://github.com/conda/conda/issues/11893 :D That's exactly the problem: some `sh` implementations absorb PS1.

**Comment by ebebpl** [comment]
Duplicate issue #11942

**Comment by nottwy** [comment]
I believe this disruption would take lots of people into a big confusion, because lots of peopel who don't understand the rationale of conda too much would take this strange change of the prompt symbol as an indication of corruption of conda. 
Fortunately, I don't need to reinstall my conda. The environment I construct in the conda is really complicated and I can't imagine the situation I uninstall the current conda and reinstall all things. The pressure pushes me to spend lots of time to learn how conda initiates itself and it brings me here.

**Comment by lpilz** [comment]
I have the same issue, however `conda init bash` errored for me (see #11510)

**Comment by KitchenTable99** [comment]
Chiming in to report the same issue, the fix identified `conda init bash` resolved the issue.

**Comment by matifali** [comment]
> Chiming in to report the same issue, the fix identified `conda init bash` resolved the issue.

Does it also work inside a `Dockerfile`?

**Comment by geometrylearner** [comment]
I experience the same issue in case of **zsh**, `conda init zsh` fixed it.

**Comment by ilyasdc** [comment]
Same issue, same fix `conda init bash`, thanks @SimonPeatman !

**Comment by teshomem** [comment]
`conda init bash` doesn't fix it for me. It throws an error
`# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/exceptions.py", line 1129, in __call__
        return func(*args, **kwargs)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/cli/main.py", line 86, in main_subshell
        exit_code = do_call(args, p)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/cli/conda_argparse.py", line 93, in do_call
        return getattr(module, func_name)(args, parser)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/cli/main_init.py", line 50, in execute
        return initialize(context.conda_prefix, selected_shells, for_user, args.system,
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/core/initialize.py", line 119, in initialize
        run_plan_elevated(plan2)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/core/initialize.py", line 688, in run_plan_elevated
        result = subprocess_call(
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/site-packages/conda/gateways/subprocess.py", line 100, in subprocess_call
        stdout, stderr = process.communicate(input=stdin)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/subprocess.py", line 1024, in communicate
        stdout, stderr = self._communicate(input, endtime, timeout)
      File "/cluster/software/Miniconda3/4.9.2/lib/python3.8/subprocess.py", line 1846, in _communicate
        input_view = memoryview(self._input)
    TypeError: memoryview: a bytes-like object is required, not 'str'

`$ /cluster/software/Miniconda3/4.9.2/bin/conda init bash``

**Comment by jaimergp** [comment]
Hi @teshomem,

That error looks more like a problem with the workaround for permission elevation. Can you please file a separate issue? Thanks!

You can also try `sudo $(which conda) init --dry-run bash` in the meantime to check if permissions are indeed the problem. If you don't have sudo access, contact your HPC admin with a link to this issue.

Thanks again!

**Comment by lpilz** [comment]
@jaimergp I actually have exactly the same problem, but I think the issue #11530 already covers it @teshomem

**Comment by ndonyapour** [comment]
Hello, 
I have conda 22.11.1 and non of the solutions here worked for me.  Here is the output of 
`PS1="${PS1:-}" conda shell.posix activate` command. 
```
export CONDA_SHLVL='1'
export CONDA_PROMPT_MODIFIER='

**Comment by awr-paul** [comment]
I can confirm this is an issue, but the behaviour depends on the distribution.  For me 'conda activate' works fine with ubuntu, debian and oracle, but on Amazon linux env_prompt completely replaces the existing prompt instead of prepending to it as you would expect.

I think that strictly speaking PS1 is a reserved shell variable so therefore not an environment variable which is exported.  Therefore, I suspect that expecting this to return the existing prompt in conda/activate.py is in fact the bug:
```python
ps1 = self.environ.get("PS1", "")
```

**Comment by jaimergp** [comment]
Is the problem fixed if you manually `export PS1=${PS1:-}` in your bashrc (or equivalent) before the `conda initialize` block?

**Comment by mirjunaid26** [comment]
> same here, `conda init bash` helped me out.
> 
> steps to reproduce: shell: zsh
> 
> 1. `./Downloads/Miniconda3-latest-Linux-x86_64.sh`
> 2. `conda config --set auto_activate_base false`
> 3. `conda update -n base -c defaults conda`
> 4. restart shell
> 5. `conda activate base` (at this step I got broken PS1
> 6. running: `conda init bash` fixed issue.

`conda init bash` couldn't solve my problem but following this procedure fixed the issue.

**Comment by jdblischak** [comment]
For anyone else that needs to fix this for conda 22.9.0, I applied the workaround from @jaimergp in https://github.com/conda/conda/issues/11885#issuecomment-1268585462 using `sed`, where `PREFIX` is the installation prefix, the same as used by the installer

```sh
for CONDAFILE in condabin/conda bin/conda bin/conda-env
do
  sed -i -e "1i #!$PREFIX/bin/python" -e '1,3d' "$PREFIX/$CONDAFILE"
done
```

Note that I'm stuck on conda 22.9.0 because my HPC requires openssl v1 to use its custom SSL certificates, and the latest Mambaforge installer that installs openssl v1 is [22.9.0-1](https://github.com/conda-forge/miniforge/releases/tag/22.9.0-1). Of course if you can install 22.11.0, you should just do that instead of applying this patch
